internal/deploy/docker.go

```diff
@@ -92,13 +112,61 @@ func (c *dockerDeploymentManifest) Inspect(service string) (*ServiceManifest, er
 		ID:         inspect.ID,
 		Name:       strings.TrimPrefix(inspect.Name, "/"),
 		Connection: service,
-		Hostname:   inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Alias:      inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Hostname:   inspect.Config.Hostname,
+		Platform:   inspect.Platform,
 	}, nil
 }
 
+// Mount will mount a service with ability to perform actions within that services environment
+// TODO: Not a fan of passing in installType here, should think about abstracting that portion out more
+func (c *dockerDeploymentManifest) Mount(service string, installType string) (installer.Package, error) {
+	log.WithFields(log.Fields{
+		"service":     service,
+		"installType": installType,
+	}).Trace("Mounting service for configuration")
+
+	container, _ := c.Inspect(service)
+	var install installer.Package
+	if strings.EqualFold(service, "elastic-agent") {
+		switch installType {
+		case "tar":
+			install = installer.NewElasticAgentTARPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		case "rpm":
+			install = installer.NewElasticAgentRPMPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		case "deb":
+			install = installer.NewElasticAgentDEBPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		}
+	}
+
+	return nil, nil
+}
+
 // Remove remove services from deployment
 func (c *dockerDeploymentManifest) Remove(services []string, env map[string]string) error {
-	serviceManager := compose.NewServiceManager()
+	for _, service := range services[1:] {
+		manifest, _ := c.Inspect(service)
+		_, err := shell.Execute(c.Context, ".", "docker", "rm", "-f", manifest.Name)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Start a container
+func (c *dockerDeploymentManifest) Start(service string) error {
+	manifest, _ := c.Inspect(service)
+	_, err := shell.Execute(c.Context, ".", "docker", "start", manifest.Name)
+	return err
+}
 
-	return serviceManager.RemoveServicesFromCompose(c.Context, services[0], services[1:], env)
```

Comment:
We are moving from docker-compose to docker more and more, right? At least for the start/stop/remove operations I think it makes sense, as it simplifies the operations. OTOH, if there is something we can improve in the docker-compose APIs from testcontainers, let's improve it :)

Reply:
This was actually a stop gap as I couldn't get this function to work, I'm going to revisit it before this PR is ready for merge. But yea I think you're right, we seem to be moving more and more to docker to start/stop/remove (and even execing). We still need testcontainers for everything else and I want to rely on it as much as possible